File: app/src/main/java/com/druger/firstandroidgame/BotBorder.java
package com.druger.firstandroidgame;

import android.graphics.Bitmap;
import android.graphics.Canvas;

/**
 * Created by druger on 12.08.2015.
 */
public class BotBorder extends GameObject {
    private Bitmap image;

    public BotBorder(Bitmap res, int x, int y) {
        width = 20;
        height = 200;

        this.x = x;
        this.y = y;
        dx = GamePanel.MOVESPEED;

        image = Bitmap.createBitmap(res, 0, 0, width, height);
    }

    public void update(){
        x += dx;
    }

    public void draw(Canvas canvas){
        try {
            canvas.drawBitmap(image, x, y, null);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/druger/firstandroidgame/objects/BotBorder.kt
package com.druger.firstandroidgame.objects;

import android.graphics.Bitmap
import android.graphics.Canvas
import com.druger.firstandroidgame.view.GamePanel

/**
 * Created by druger on 12.08.2015.
 */
class BotBorder(res: Bitmap, x: Int, y: Int) : GameObject() {
    private var image: Bitmap

    init {
        width = 20
        height = 200
        this.x = x
        this.y = y
        dx = GamePanel.MOVESPEED
        image = Bitmap.createBitmap(res, 0, 0, width, height)
    }

    fun update() {
        x += dx;
    }

    fun draw(canvas: Canvas) {
        try {
            canvas.drawBitmap(image, x.toFloat(), y.toFloat(), null);
        } catch (e: Exception) {
            e.printStackTrace();
        }
    }
}
